lib/darlingtonia/validators/csv_format_validator.rb

```diff
@@ -2,7 +2,23 @@
 
 module Darlingtonia
   ##
-  # A validator for correctly formatted CSV
+  # A validator for correctly formatted CSV.
+  #
+  # @example
+  #   parser = Parser.new(file: File.open('path/to/my.csv'))
+  #
+  #   CsvFormatValidator.new.validate(parser: parser)
+  #
+  # @see http://ruby-doc.org/stdlib-2.0.0/libdoc/csv/rdoc/CSV/MalformedCSVError.html
   class CsvFormatValidator < Validator
+    ##
+    # @see Validator#validate
+    def validate(parser:, **)
+      return [] if CSV.parse(parser.file.read)
+    rescue CSV::MalformedCSVError => e
```

Comment:
Does the rescue just catch the first error it comes across?

Reply:
It catches errors of class `CSV::MalformedCSVError`, which is what Ruby's `CSV` parser raises for any ill-formed data.

We don't reraise, here, because we want to just return a formatted error collection.